File: Upcoder/Other/TINH_TONG/LTCB-6.cpp
#include <bits/stdc++.h>
using namespace std;
double DequyS(int x, int n)
{
    if (n == 0)
        return 0;
    return 1.0 / (n * x * (n + 1)) + DequyS(x, n - 1);
}
int main()
{
    int x, n;
    cin >> x >> n;
    if(x < 0 or n < 0)
        cout << -1;
    else
        cout << fixed << setprecision(3) << DequyS(x, n);
    return 0;
}

File: Upcoder/Other/TINH_TONG/LTCB-6.py
def DequyS(x, n):
    if n == 0:
        return 0
    else:
        return 1.0 / (n * x * (n + 1)) + DequyS(x, n - 1)

# Nhập giá trị x và n
x, n = map(int, input().split())

# Tính giá trị s
if x < 0 or n < 0:
    print(-1)
else:
    s = DequyS(x, n)
    print(f"{s:.3f}")
